plugin/src/main/java/org/opensearch/ml/sdkclient/LocalClusterIndicesClient.java

```diff
@@ -158,6 +136,10 @@ public CompletionStage<UpdateDataObjectResponse> updateDataObjectAsync(UpdateDat
                         new UpdateRequest(request.index(), request.id()).doc(request.dataObject().toXContent(sourceBuilder, EMPTY_PARAMS))
                     )
                     .actionGet();
+                if (updateResponse == null) {
+                    log.info("Null UpdateResponse");
+                    return new UpdateDataObjectResponse.Builder().id(request.id()).parser(null).build();
```

Comment:
Is this expected? Should we throw error?

Reply:
This (and a similar one for Get) I kept to keep existing behavior/tests.  I spent nearly an hour digging through the NodeClient code to see when this might happen and do not see how we would ever get a null response however... other than as generated in tests, but the code passes along the null faithfully.